api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -44,13 +47,26 @@ public FileIteratingFirehose(
   {
     this.lineIterators = lineIterators;
     this.parser = parser;
+    final ParseSpec parseSpec = parser.getParseSpec();
+    if (parseSpec instanceof CSVParseSpec) {
```

Comment:
Ah, yeah. I was thinking that ParseSpec should describe the behavior of the parser which should be consistent no matter what the type of index task is. It would be more intuitive and reduce user mistakes. Even though ParseSpec is in druid-api, I think we need to support a consistent behavior if it is provided by default. What do you think?